### PR TITLE
Use Java 9+ methods in `JavaVMArguments`

### PR DESCRIPTION
**Pull request by basil**
Avoid the fragile `ProcessTree` code if we can avoid it by relying on native Java 9+ functionality. I deemed this too trivial for a changelog entry.

### Testing done

Stepped through the new code in a debugger to ensure it was working as intended. Started Jenkins on Linux with `java -jar jenkins.war`, installed a plugin, and clicked on "Restart Jenkins when installation is complete and no jobs are running". Verified that Jenkins restarted successfully.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6758"></a>

### CODE REVIEW

File: core/src/main/java/jenkins/util/JavaVMArguments.java
Comment (timja) [inline core/src/main/java/jenkins/util/JavaVMArguments.java:31]: is there a reason the native approach has been kept? both here and in the restarter?

Comment (basil) [reply core/src/main/java/jenkins/util/JavaVMArguments.java:31]: I should hope there is a good reason. I would not want to do anything based on sloppy reasoning. While implementing `ProcessHandle.pid()` merely requires a call to `getpid(2)`, there is no standard C or C++ API that could be used to implement `ProcessHandle.info()`: it requires OS-specific code, some of which can be fairly complicated even in C/C++ (and even more so in Java, as can be seen from our `ProcessTree` class). The documentation for [`ProcessHandle.Info`](https://docs.oracle.com/javase/9/docs/api/java/lang/ProcessHandle.Info.html) returns optional types for everything and is very vague about what is supported:

> The attributes of a process vary by operating system and are not available in all implementations. Information about processes is limited by the operating system privileges of the process making the request. The return types are `Optional<T>` allowing explicit tests and actions if the value is available.

While in practice this API has returned valid results during all of my testing on Linux, I would not be surprised if the OpenJDK developers declined to implement this API for some obscure platforms like AIX or Solaris/illumos: platforms which are supported by our `ProcessTree` API, have active users who have complained when we tried to remove support for those platforms, and whose users would be negatively affected by any potential regression caused by the removal of our (working!) `ProcessTree` implementation without a suitable replacement from OpenJDK. The basic point is that the API explicitly states that it cannot be relied on, so we do not rely on it.

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
